src/v/raft/tests/raft_fixture.cc

```diff
@@ -281,8 +292,24 @@ in_memory_test_protocol::dispatch(model::node_id id, ReqT req) {
     iobuf buffer;
     co_await serde::write_async(buffer, std::move(req));
     try {
-        auto resp = co_await node_channel.exchange(
-          map_msg_type<ReqT>(), std::move(buffer));
+        const auto msg_type = map_msg_type<ReqT>();
+
+        if (auto iter = _failures.find(msg_type); iter != _failures.end()) {
+            auto& fail = iter->second;
+            co_await ss::visit(fail, [this, msg_type](response_delay& f) {
```

Comment:
nit: i would probably use a command pattern here i.e. instead of a visit encapsulate the failure injection logic inside the operation itself, but it is a matter of taste